modules/mono/editor/GodotTools/GodotTools/BottomPanel.cs

```diff
@@ -24,48 +24,50 @@ public class BottomPanel : VBoxContainer
         private Button errorsBtn;
         private Button viewLogBtn;
 
-        private void _UpdateBuildTabsList()
+        private void _UpdateBuildTab(int index, int? currentTab)
         {
-            buildTabsList.Clear();
+            var tab = (BuildTab)buildTabs.GetChild(index);
 
-            int currentTab = buildTabs.CurrentTab;
+            string itemName = Path.GetFileNameWithoutExtension(tab.BuildInfo.Solution);
+            itemName += " [" + tab.BuildInfo.Configuration + "]";
 
-            bool noCurrentTab = currentTab < 0 || currentTab >= buildTabs.GetTabCount();
+            buildTabsList.AddItem(itemName, tab.IconTexture);
 
-            for (int i = 0; i < buildTabs.GetChildCount(); i++)
-            {
-                var tab = (BuildTab)buildTabs.GetChild(i);
+            string itemTooltip = "Solution: " + tab.BuildInfo.Solution;
+            itemTooltip += "\nConfiguration: " + tab.BuildInfo.Configuration;
+            itemTooltip += "\nStatus: ";
 
-                if (tab == null)
-                    continue;
+            if (tab.BuildExited)
+                itemTooltip += tab.BuildResult == BuildTab.BuildResults.Success ? "Succeeded" : "Errored";
+            else
+                itemTooltip += "Running";
 
-                string itemName = Path.GetFileNameWithoutExtension(tab.BuildInfo.Solution);
-                itemName += " [" + tab.BuildInfo.Configuration + "]";
+            if (!tab.BuildExited || tab.BuildResult == BuildTab.BuildResults.Error)
+                itemTooltip += $"\nErrors: {tab.ErrorCount}";
 
-                buildTabsList.AddItem(itemName, tab.IconTexture);
+            itemTooltip += $"\nWarnings: {tab.WarningCount}";
 
-                string itemTooltip = "Solution: " + tab.BuildInfo.Solution;
-                itemTooltip += "\nConfiguration: " + tab.BuildInfo.Configuration;
-                itemTooltip += "\nStatus: ";
+            buildTabsList.SetItemTooltip(index, itemTooltip);
 
-                if (tab.BuildExited)
-                    itemTooltip += tab.BuildResult == BuildTab.BuildResults.Success ? "Succeeded" : "Errored";
-                else
-                    itemTooltip += "Running";
+            // If this tab was already selected before the changes or if no tab was selected
+            if (currentTab == null || currentTab == index)
+            {
+                buildTabsList.Select(index);
+                _BuildTabsItemSelected(index);
+            }
+        }
 
-                if (!tab.BuildExited || tab.BuildResult == BuildTab.BuildResults.Error)
-                    itemTooltip += $"\nErrors: {tab.ErrorCount}";
+        private void _UpdateBuildTabsList()
+        {
+            buildTabsList.Clear();
 
-                itemTooltip += $"\nWarnings: {tab.WarningCount}";
+            int? currentTab = buildTabs.CurrentTab;
```

Comment:
`buildTabs.CurrentTab` is actually of type `int` not `int?`, so the only `null` case is the below assignment.

Reply:
If no build tab is selected or the selected build tab no longer exists we set it to `null` below.